src/plugins/orangehrmRecruitmentPlugin/Dao/RecruitmentAttachmentDao.php

```diff
@@ -129,14 +141,13 @@ public function getVacancyAttachmentsByVacancyId(int $vacancyId): array
      * @param int $vacancyId
      * @return int
      */
-    public function getAttachmentCount(int $vacancyId): int
+    public function getVacancyAttachmentsCountByVacancyId(int $vacancyId): int
     {
         $qb = $this->createQueryBuilder(VacancyAttachment::class, 'attachment');
```

Comment:
```php
        $qb = $this->createQueryBuilder(VacancyAttachment::class, 'attachment');
        $qb->where('attachment.vacancy = :vacancyId');
        $qb->setParameter('vacancyId', $vacancyId);
        return $this->count($qb);
```
this should work. try to avoid table joins